diff --git a/recipes/libtasn1.recipe b/recipes/libtasn1.recipe
index 91b44c4..7da2137 100644
--- a/recipes/libtasn1.recipe
+++ b/recipes/libtasn1.recipe
@@ -8,7 +8,10 @@ class Recipe(recipe.Recipe):
     tarball_checksum = '7e528e8c317ddd156230c4e31d082cd13e7ddeb7a54824be82632209550c8cca'
     licenses = [License.LGPLv2_1Plus]
     autoreconf = True
-    patches = ['libtasn1/0001-Rename-gnulib-symbols.patch']
+    patches = [
+        'libtasn1/0001-Rename-gnulib-symbols.patch',
+        'libtasn1/0002-Android-opaque-FILE.patch',
+    ]
     configure_options = '--disable-doc'
 
     files_libs = ['libtasn1']
