diff --git a/recipes/wpewebkit.recipe b/recipes/wpewebkit.recipe
index 92a9220..87d9134 100644
--- a/recipes/wpewebkit.recipe
+++ b/recipes/wpewebkit.recipe
@@ -45,6 +45,7 @@ class Recipe(recipe.Recipe):
     patches = [
         'wpewebkit/0001-WPE-Decouple-touch-driven-scrolling-unconditionally.patch',
         'wpewebkit/0002-KaiOS-Android-use-GioUnix.patch',
+        'wpewebkit/0003-KaiOS-Android-ARMv7-JIT.patch',
     ]
 
     files_libs = [
