diff --git a/recipes/openssl.recipe b/recipes/openssl.recipe
index 31d8d59..c33b1a4 100644
--- a/recipes/openssl.recipe
+++ b/recipes/openssl.recipe
@@ -8,11 +8,11 @@ class Recipe(recipe.Recipe):
     # Note: openssl helpfully moves tarballs somewhere else (old/x.y.z/)
     # whenever a new release comes out, so make sure to mirror to fdo when
     # bumping the release!
-    version = '1.1.1s'
-    licenses = [{License.OPENSSL: ['LICENSE']}]
+    version = '3.5.7'
+    licenses = [License.Apachev2]
     stype = SourceType.TARBALL
-    url = 'https://github.com/openssl/openssl/releases/download/OpenSSL_1_1_1s/{0}-{1}.tar.gz'.format(name, version)
-    tarball_checksum = 'c5ac01e760ee6ff0dab61d6b2bbd30146724d063eb322180c6f18a6f74e4b6aa'
+    url = 'https://github.com/openssl/openssl/releases/download/openssl-%(version)s/openssl-%(version)s.tar.gz'
+    tarball_checksum = 'a8c0d28a529ca480f9f36cf5792e2cd21984552a3c8e4aa11a24aa31aeac98e8'
     deps = ['ca-certificates', 'zlib']
     # Parallel make fails randomly due to undefined macros, probably races
     allow_parallel_build = False
@@ -21,20 +21,7 @@ class Recipe(recipe.Recipe):
     # Can build for MSVC and UWP
     can_msvc = True
 
-    patches = [
-        # Portable prefix with SSL certs
-        'openssl/0001-Load-ca-certificate.crt-from-PREFIX-etc-ssl-on-macOS.patch',
-        # MSVC and UWP support
-        'openssl/0001-windows-makefile.tmpl-Generate-and-install-pkgconfig.patch',
-        'openssl/0002-windows-makefile.tmpl-Fix-ONECORE-build.patch',
-        'openssl/0003-windows-makefile.tmpl-Do-not-prefix-import-libraries.patch',
-        # https://github.com/openssl/openssl/pull/12400
-        'openssl/0001-crypto-win-Don-t-use-disallowed-APIs-on-UWP.patch',
-        'openssl/0002-win-onecore-Build-with-APPCONTAINER-for-UWP-compat.patch',
-        # Android support
-        'openssl/0003-openssl-Support-loading-ca-certificates-on-Android.patch',
-    ]
+    patches = []
 
     files_bins = ['openssl']
     files_libs = ['libcrypto', 'libssl']
@@ -192,6 +179,7 @@ class Recipe(recipe.Recipe):
         if self.config.target_platform == Platform.ANDROID:
             self.prepend_env('PATH', self.get_env('ANDROID_NDK_TOOLCHAIN_BIN'), sep=os.pathsep)
+            self.set_env('ANDROID_NDK_ROOT', self.get_env('ANDROID_NDK_HOME'))
 
 
     @modify_environment
@@ -207,7 +195,8 @@ class Recipe(recipe.Recipe):
         # OpenSSL guesses the libdir incorrectly on x86_64
         config_sh = 'perl ./Configure --prefix=' + self.config.prefix + \
-            ' --libdir=lib' + self.config.lib_suffix + '  no-makedepend no-unit-test '
+            ' --libdir=lib' + self.config.lib_suffix + \
+            ' no-makedepend no-unit-test no-tests no-fuzz-afl no-fuzz-libfuzzer no-h3demo '
         if self.config.target_platform == Platform.IOS:
             # Note: disable 'no-devcryptoeng' when we finally target the
             # *real* ios configuration targets
