diff --git a/recipes/wpewebkit.recipe b/recipes/wpewebkit.recipe
index 4430b875..2ea9b9ff 100644
--- a/recipes/wpewebkit.recipe
+++ b/recipes/wpewebkit.recipe
@@ -32,8 +32,7 @@ class Recipe(recipe.Recipe):
         'libxslt',
         'libwebp',
         'libwpe',
-        'openjpeg',
-        'openxr',
+        'xkbcommon',
     ]
     # FIXME: support accessibility, woff2, hyphenation
     #
@@ -45,46 +44,20 @@ class Recipe(recipe.Recipe):
     # proper fix for this.
     configure_options = '\
         -DPORT=WPE \
-        -DUSE_GSTREAMER_GL=ON \
-        -DUSE_WOFF2=OFF \
-        -DENABLE_DOCUMENTATION=OFF \
-        -DENABLE_INTROSPECTION=OFF \
-        -DENABLE_JOURNALD_LOG=OFF \
-        -DENABLE_BUBBLEWRAP_SANDBOX=OFF \
-        -DENABLE_SPEECH_SYNTHESIS=OFF \
-        -DENABLE_SPELLCHECK=OFF \
-        -DENABLE_WEBXR=ON \
-        -DENABLE_WEBXR_LAYERS=ON \
-        -DENABLE_WEBXR_HIT_TEST=ON \
-        -DUSE_ATK=OFF \
-        -DUSE_LCMS=OFF \
-        -DUSE_AVIF=OFF \
-        -DUSE_GBM=OFF \
-        -DUSE_LIBHYPHEN=OFF \
-        -DUSE_LIBDRM=OFF \
-        -DUSE_JPEGXL=OFF \
-        -DUSE_LIBBACKTRACE=OFF \
-        -DUSE_SKIA_OPENTYPE_SVG=OFF \
-        -DUSE_SYSPROF_CAPTURE=OFF \
-        -DENABLE_WEBDRIVER=ON \
-        -DENABLE_WPE_PLATFORM=ON \
-        -DENABLE_WPE_PLATFORM_DRM=OFF \
-        -DENABLE_WPE_PLATFORM_WAYLAND=OFF \
-        -DENABLE_WPE_PLATFORM_HEADLESS=ON \
-        -DENABLE_SWIFT_DEMO_URI_SCHEME=OFF \
-        -DENABLE_BACK_FORWARD_LIST_SWIFT=OFF \
         -DCMAKE_BUILD_TYPE=Release \
         -DCMAKE_DISABLE_PRECOMPILE_HEADERS=ON \
+        -DCMAKE_CXX_FLAGS=-I%(prefix)s/include/gio-unix-2.0 \
     '
+    configure_options += ' ' + os.environ['OOS_WPE_FEATURE_OPTIONS']
     cmake_generator = 'ninja'
 
     patches = [
         'wpewebkit/0001-WPE-Decouple-touch-driven-scrolling-unconditionally.patch',
+        'wpewebkit/0002-KaiOS-Android-use-GioUnix.patch',
     ]
 
     files_libs = [
         'libWPEWebKit-2.0',
-        'libWPEWebDriver'
     ]
 
     files_misc = [
