diff --git a/recipes/wpewebkit.recipe b/recipes/wpewebkit.recipe
index d536d7cb..3204df94 100644
--- a/recipes/wpewebkit.recipe
+++ b/recipes/wpewebkit.recipe
@@ -63,6 +63,7 @@ class Recipe(recipe.Recipe):
         'wpewebkit/0005-OOS-context-menus-off-build.patch',
         'wpewebkit/0006-OOS-libdrm-off-build.patch',
         'wpewebkit/0007-OOS-context-menus-off-link-build.patch',
+        'wpewebkit/0008-OOS-eager-Wasm-BBQ.patch',
     ]
 
     files_libs = [
