diff --git a/recipes/wpewebkit.recipe b/recipes/wpewebkit.recipe
index e0c2db43..d536d7cb 100644
--- a/recipes/wpewebkit.recipe
+++ b/recipes/wpewebkit.recipe
@@ -62,6 +62,7 @@ class Recipe(recipe.Recipe):
         'wpewebkit/0004-KaiOS-Android23-legacy-buffer.patch',
         'wpewebkit/0005-OOS-context-menus-off-build.patch',
         'wpewebkit/0006-OOS-libdrm-off-build.patch',
+        'wpewebkit/0007-OOS-context-menus-off-link-build.patch',
     ]
 
     files_libs = [
