diff --git a/recipes/gst-plugins-bad-1.0.recipe b/recipes/gst-plugins-bad-1.0.recipe
index 62e950be..019494f4 100644
--- a/recipes/gst-plugins-bad-1.0.recipe
+++ b/recipes/gst-plugins-bad-1.0.recipe
@@ -22,11 +22,11 @@ class Recipe(custom.GStreamer):
                      'codectimestamper': 'enabled',
                      'dash': 'enabled',
                      'decklink': 'enabled',
-                     'dtls': 'enabled',
+                     'dtls': 'disabled',
                      'dts': 'enabled',
                      'hls': 'enabled',
                      'hls-crypto': 'openssl',
-                     'isac': 'enabled',
+                     'isac': 'disabled',
                      'ivfparse': 'enabled',
                      'ladspa': 'auto', # lrdf is optional
                      'openh264': 'enabled',
@@ -35,19 +35,19 @@ class Recipe(custom.GStreamer):
                      'rtmp': 'enabled',
                      'rtmp2': 'enabled',
                      'sbc': 'enabled',
-                     'sctp': 'enabled',
-                     'sctp-internal-usrsctp': 'enabled',
+                     'sctp': 'disabled',
+                     'sctp-internal-usrsctp': 'disabled',
                      'soundtouch': 'enabled',
                      'srt': 'enabled',
                      'spandsp': 'enabled',
-                     'srtp': 'enabled',
+                     'srtp': 'disabled',
                      'resindvd': 'enabled',
                      'rist' : 'enabled',
                      'rsvg': 'enabled',
                      'voaacenc': 'enabled',
                      'vulkan': 'enabled',
-                     'webrtc': 'enabled',
-                     'webrtcdsp': 'enabled',
+                     'webrtc': 'disabled',
+                     'webrtcdsp': 'disabled',
                      'zbar': 'enabled',
                      # disabled plugins
                      'amfcodec': 'disabled',
@@ -84,9 +84,9 @@ class Recipe(custom.GStreamer):
                     }
     deps = ['gstreamer-1.0', 'gst-plugins-base-1.0', 'bzip2', 'libass',
             'zlib', 'openh264', 'opus', 'librtmp', 'libxml2',
-            'libsrtp', 'libdca', 'libdvdnav', 'libnice',
+            'libdca', 'libdvdnav',
             'soundtouch', 'vo-aacenc', 'librsvg', 'openjpeg',
-            'pango', 'spandsp', 'webrtc-audio-processing', 'sbc', 'ladspa',
+            'pango', 'spandsp', 'sbc', 'ladspa',
             'srt', 'zbar', 'libltc']
     use_system_libs = True
