diff --git a/recipes/wpewebkit.recipe b/recipes/wpewebkit.recipe
index 5177e489..37a486ac 100644
--- a/recipes/wpewebkit.recipe
+++ b/recipes/wpewebkit.recipe
@@ -60,6 +60,7 @@ class Recipe(recipe.Recipe):
         'wpewebkit/0002-KaiOS-Android-use-GioUnix.patch',
         'wpewebkit/0003-KaiOS-Android-ARMv7-JIT.patch',
         'wpewebkit/0004-KaiOS-Android23-legacy-buffer.patch',
+        'wpewebkit/0005-OOS-context-menus-off-build.patch',
     ]
 
     files_libs = [
