diff --git a/recipes/wpewebkit.recipe b/recipes/wpewebkit.recipe
index 37a486ac..e0c2db43 100644
--- a/recipes/wpewebkit.recipe
+++ b/recipes/wpewebkit.recipe
@@ -61,6 +61,7 @@ class Recipe(recipe.Recipe):
         'wpewebkit/0003-KaiOS-Android-ARMv7-JIT.patch',
         'wpewebkit/0004-KaiOS-Android23-legacy-buffer.patch',
         'wpewebkit/0005-OOS-context-menus-off-build.patch',
+        'wpewebkit/0006-OOS-libdrm-off-build.patch',
     ]
 
     files_libs = [
