diff --git a/cerbero/bootstrap/android.py b/cerbero/bootstrap/android.py
--- a/cerbero/bootstrap/android.py
+++ b/cerbero/bootstrap/android.py
@@ -37,6 +37,8 @@ class AndroidBootstrapper(BootstrapperBase):
     def __init__(self, config, offline, assume_yes):
         super().__init__(config, offline, 'android')
         self.prefix = self.config.toolchain_prefix
+        if os.path.isfile(os.path.join(self.prefix, 'source.properties')):
+            return
         url = NDK_BASE_URL % (NDK_VERSION, self.config.platform)
         self.fetch_urls.append((url, None, NDK_CHECKSUMS[os.path.basename(url)]))
         self.extract_steps.append((url, True, self.prefix))
