diff --git a/recipes/wpebackend-android.recipe b/recipes/wpebackend-android.recipe
--- a/recipes/wpebackend-android.recipe
+++ b/recipes/wpebackend-android.recipe
@@ -14,6 +14,7 @@ class Recipe(recipe.Recipe):
 
     patches = [
         'wpebackend-android/0001-OOS-RGB565-and-Android23-buffer.patch',
+        'wpebackend-android/0002-OOS-seqpacket-renderer-IPC.patch',
     ]
 
     files_libs = ['libWPEBackend-android']
