diff --git a/src/renderer-backend-egl.cpp b/src/renderer-backend-egl.cpp
index ca8002d..638ea78 100644
--- a/src/renderer-backend-egl.cpp
+++ b/src/renderer-backend-egl.cpp
@@ -335,6 +335,6 @@ void EGLTarget::frameWillRender()
         description.width = renderer.width;
         description.height = renderer.height;
         description.layers = 1;
-        description.format = AHARDWAREBUFFER_FORMAT_R8G8B8A8_UNORM;
+        description.format = AHARDWAREBUFFER_FORMAT_R5G6B5_UNORM;
         description.usage = AHARDWAREBUFFER_USAGE_GPU_FRAMEBUFFER | AHARDWAREBUFFER_USAGE_GPU_SAMPLED_IMAGE | AHARDWAREBUFFER_USAGE_COMPOSER_OVERLAY;
         description.stride = description.rfu0 = description.rfu1 = 0;
