diff --git a/Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp b/Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp
index 352281a21a1b..1f950542432b 100644
--- a/Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp
+++ b/Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp
@@ -739,10 +739,15 @@ auto MediaControlsHost::mediaControlsContextMenuItems(String&& optionsJSONString
     ASSERT(!idMap.isEmpty());
 
     return std::make_pair(WTF::move(items), WTF::move(idMap));
+#else
+    UNUSED_PARAM(optionsJSONString);
+    return { };
+#endif
 }
 
 bool MediaControlsHost::showMediaControlsContextMenu(HTMLElement& target, String&& optionsJSONString, Ref<VoidCallback>&& callback)
 {
+#if USE(UICONTEXTMENU) || (ENABLE(CONTEXT_MENUS) && USE(ACCESSIBILITY_CONTEXT_MENUS))
     Ref mediaElement = m_mediaElement.get();
     RefPtr page = mediaElement->document().page();
     if (!page)
@@ -853,6 +858,9 @@ bool MediaControlsHost::showMediaControlsContextMenu(HTMLElement& target, String
 
     return true;
 #else // USE(UICONTEXTMENU) || (ENABLE(CONTEXT_MENUS) && USE(ACCESSIBILITY_CONTEXT_MENUS))
+    UNUSED_PARAM(target);
+    UNUSED_PARAM(optionsJSONString);
+    UNUSED_PARAM(callback);
     return { };
 #endif
 }
